nomad/secure_variables_endpoint.go

```diff
@@ -367,3 +367,56 @@ func (sv *SecureVariables) decrypt(v *structs.SecureVariable) error {
 	v.EncryptedData = nil
 	return nil
 }
+
+// handleMixedAuthEndpoint is a helper to handle auth on RPC endpoints that can
+// either be called by external clients or by workload identity
+func (sv *SecureVariables) handleMixedAuthEndpoint(args structs.QueryOptions, cap, pathOrPrefix string) error {
+
+	// Perform the initial token resolution.
+	aclObj, err := sv.srv.ResolveToken(args.AuthToken)
+	if err == nil {
+		// Perform our ACL validation. If the object is nil, this means ACLs
+		// are not enabled, otherwise trigger the allowed namespace function.
+		if aclObj != nil {
+			if !aclObj.AllowNsOp(args.RequestNamespace(), cap) {
+				return structs.ErrPermissionDenied
+			}
+		}
+		return nil
+	}
+
```

Comment:
We may want to only continue when `err == ErrTokenNotFound`? We should at least log when `err != ErrTokenNotFound` as otherwise we'd be silently ignoring some error conditions that indicate pretty weird states (such as being unable to compile policies). I also don't want to assume any future error return values in the `Server.ResolveToken` path are safe to silently drop.

Reply:
Good catch! Will fix.

Reply:
Resolved in 23af20e29b2398437d86c45487d7bccf7f639cb3